src/app/view/components/_common/push-publish-dialog/push-publish-dialog.component.ts

```diff
@@ -0,0 +1,121 @@
+import { BehaviorSubject } from 'rxjs/Rx';
+import { Component, Input, Output, EventEmitter, ViewEncapsulation } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { OnInit } from '@angular/core/src/metadata/lifecycle_hooks';
+import { PushPublishService } from '../../../../api/services/push-publish/push-publish.service';
+import { SelectItem } from 'primeng/primeng';
+import { DotMessageService } from '../../../../api/services/dot-messages-service';
+
+@Component({
+    encapsulation: ViewEncapsulation.None,
+    selector: 'dot-push-publish-dialog',
+    styleUrls: ['./push-publish-dialog.component.scss'],
+    templateUrl: 'push-publish-dialog.component.html'
+})
+
+export class PushPublishContentTypesDialogComponent implements OnInit {
+    publishdate: Date;
+    expiredate: Date;
+    form: FormGroup;
+    errorMessage: BehaviorSubject<string> = new BehaviorSubject(null);
+    forcePushCheckbox = false;
+    pushActions: SelectItem[];
+    publish = true;
+    showExpireDate = false;
+    submitted = false;
+    @Input() displayDialog = false;
+    @Input() contentTypeId: string;
+    @Output() cancel = new EventEmitter<boolean>();
+
+    constructor(
+        private pushPublishService: PushPublishService,
+        public fb: FormBuilder,
+        public dotMessageService: DotMessageService
+    ) {}
+
+    ngOnInit() {
+        this.pushActions = [
+            { label: 'Push', value: 'publish' },
+            { label: 'Remove', value: 'expire' },
+            { label: 'Push Expire', value: 'publishexpire' }
```

Comment:
maybe we should create a enum with these values: 'publish', 'expire' and 'publishexpire'